### ISSUE TITLE
Map not working

### ISSUE THREAD

**Issue by Wolfie57** [jghazally/quakemonitor#1]
Hi jghazally
Recent eq activity got me looking, found your script, but have an issue.
Map won't kick in, "Ooops" msg.  Js console hilites "TypeError: d is undefined . . . common.js:"

I have my learners PHP ticket, but JS is still a total mystery. Any thoughts?
Cheers

**Comment by evilbudz** [comment]
i to could not get to work, here is my index.php , witch is working fine now
see at https://www.nzquakes.maori.nz

<!–– use code after this <?php
include_once('config.php');

if ( !empty($_GET['ajax']) ) {
		// OK cool - then let's create a new cURL resource handle
	$ch = curl_init();

	// Now set some options (most are optional)

	// Set URL to download
	curl_setopt($ch, CURLOPT_URL, 'https://www.geonet.org.nz/quakes/services/all.json');

	// Include header in result? (0 = yes, 1 = no)
	curl_setopt($ch, CURLOPT_HEADER, 0);

	// Should cURL return or print out the data? (true = return, false = print)
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	// Timeout in seconds
	curl_setopt($ch, CURLOPT_TIMEOUT, 30);

	// Download the given URL, and return output
	$output = curl_exec($ch);

	// Close the cURL resource, and free system resources
	curl_close($ch);

	echo $output;
	die;
}
?> ––>
<!DOCTYPE html>

<html lang="en-NZ">

<html>

<head>
	<title>NZ Quakes | Earth Quakes in New Zealand</title>
	
	<meta http-equiv="refresh" content="60">
	
	<meta charset="utf-8">

	<meta http-equiv="X-UA-Compatible" content="IE=edge">

	<meta name="description" content="NZ Quakes | Earth Quakes in New Zealand">

	<meta name="keywords" content="nz quakes, nz, new zealand, christchurch, wellington, auckland, chch, quake, earth quake, tremble, shake, tremor, shake, quake drum"> 

	<meta name="viewport" content="width=device-width,initial-scale=1">

	<meta name="author" content="Timothy Vercoe">

	<meta property="og:url" content="https://www.nzquakes.maori.nz/">

	<meta property="og:description" content="NZ Quakes | Earth Quakes in New Zealand">

	<meta property="og:site_name" content="NZ Quakes | Earth Quakes in New Zealand">

	<meta property="og:type" content="website">

	<meta property="og:title" content="NZ Quakes">

	<meta property="og:image:secure_url" content="https://www.nzquakes.maori.nz/images/preview-logo.jpg">
	
	<link rel="shortcut icon" href="https://www.nzquakes.maori.nz/images/favicon.ico">
	
	<link rel="icon" type="image/png" href="https://www.nzquakes.maori.nz/images/favicon.png">
	
	<script src="Scripts/libs/jquery.js"></script>
	
	<script type="text/javascript" src="https://maps.googleapis.com/maps/api/js?key=YOUR-API-KEY-HERE"></script>
	
	<script src="https://www.nzquakes.maori.nz/Scripts/libs/gmaps.js"></script>
	
	<script src="https://www.nzquakes.maori.nz/Scripts/libs/twinkle.js"></script>
	
	<script src="https://www.nzquakes.maori.nz/Scripts/main.js"></script>
		
	<link href="https://www.nzquakes.maori.nz/css/bootstrap.css" rel="stylesheet">
	<link href="https://www.nzquakes.maori.nz/css/bootstrap-responsive.css" rel="stylesheet">
	<link href="https://www.nzquakes.maori.nz/css/screen.css" rel="stylesheet">
</head>

<body>
	<div class="container">
	<div class="page-header">
		<h1><center>NZ Quakes - Earth Quake Monitor</center></h1>
	</div>
		<div class="row">
			<div class="span12 time_since invisible">
				<center><span class="hours"></span> Hours <span class="minutes"></span> Minutes <span class="seconds"></span> Seconds since the last Earth Quake</center>
			</div>
			<div class="span6 recent_quake invisible">
				<h3>Most recent Earth Quake:<br>
					<span class="magnitude"></span>M - <span class="depth"></span> @ <span class="origintime"></span>
				</h3>
			</div>
			<div class="span6 strongest_quake">
				<h3>Strongest Earth Quake:<br>
					<span class="magnitude"></span>M - <span class="depth"></span> @ <span class="origintime"></span>
				</h3>
			</div>
		</div>
		<div id="map"></div>
	</div>
</body>
<br>
<center>Click <a href="https://www.nzquakes.maori.nz/support"  onclick="window.open(this);return false" title="Maian Support">Here</a> for Support</center>
<center>Copyright 2014 - 2018 | NZ Quakes</center>
<?php include('includes/google-ana.php'); ?>
</html>

**Comment by jghazally** [comment]
Hi @evilbudz I just checked the link: https://www.nzquakes.maori.nz/
Everything seems to be working on chrome?

**Comment by evilbudz** [comment]
sorry yes was to help that dude out i should have stated more clearly, i had to modify the script abit to get it to work but seems to work now just cant figure out how to resize the map lol its prob realy easy too

**Comment by jghazally** [comment]
What was it you had to change? Mind doing a Pull request? No worries if not. Happy it helped. Haven't looked at this in 5 years. Might have to revisit it.
If you haven't figured it out yet,
To change the zoom and centre of the map you'll need to change the map options: `Scripts/main.js`
you'll want to check the gmaps documentation [here]( http://hpneo.github.io/gmaps/)
To resize the map on the page you'll want to check the css styles: `css/screen.css`
Specifically the `#map` section.

Good luck!

**Comment by Wolfie57** [comment]
Hi Guys, "that dude" here.
I deleted my bookmark for this last week, so typical !!
Many thanks for the posts, & yep https://www.nzquakes.maori.nz/ working swell in old Firefox too.
Something to play with this weekend.
Cheers

**Comment by evilbudz** [comment]
i had to change 1 thing really that seemed important in the feild you state in index.php 

<script type="text/javascript"
src="https://maps.googleapis.com/maps/api/js?key=<?php echo GOOGLEMAPS_API ?>Q&sensor=false">
		</script>

however it should be as the above did not work until i changed this value to the actual api key as dont seem to read the config.php file

<script type="text/javascript" src="https://maps.googleapis.com/maps/api/js?key=YOUR-API-KEY-HERE"></script>

**Comment by evilbudz** [comment]
how would one add a thing to click that quake and goto the geonet or can we get this script re-done so can show relevent data for the eq clicked on ?

**Comment by evilbudz** [comment]
interesting i cannot change the value as then user has to press CTRL + scroll up and down but using 400  works anything else you have press ctrl ??
#map {
	border: 10px solid #fff;
	border-radius: 5px;
	height: 400px;
	width: 100%;
	margin-top: 30px;

**Comment by evilbudz** [comment]
mean i got everything to work now that i can see, i added a few extra things dont know how the get the name of place where eq was though any ideas

**Comment by evilbudz** [comment]
the script may work as is, but as it just does not seem to want to use the include_once for the config.php hence why i put it in manually instead of the php code to do it, maybe my hosting not allow it

**Comment by Wolfie57** [comment]
Yep, the map will display if you use a text key in place of a defined thing.
From what I found a hyphen in the key is stuffing that up, hyphens don't work in variables or defines apparently.
So 1 step forward, but still no quake data showing so far. Battling on!

**Comment by evilbudz** [comment]
email me tvercoe @ live.co.uk i will send you what i have
as i found i have had to use full web address don't know why i am guessing it is my hosting provider

**Comment by Wolfie57** [comment]
have done, cheers

**Comment by evilbudz** [comment]
did ya get the file?

**Comment by Wolfie57** [comment]
just checked & Yep, thanks bud

**Comment by evilbudz** [comment]
hey jghazally, any reason why the timer would display a - symbol in front of the time or is this a geonet error, EDIT omg what a muppet i am it was my pc date and time was changed does this all the time

**Comment by evilbudz** [comment]
would you happen to know or anyone else what the address is for the full list of quakes?
as i can only fint the all.json and the felt.json files, some website (http://www.christchurchquakemap.co.nz/faq) refrences to this full list but does not say where or how to get it, the reason i want this file is it is the full list and is not displayed on there website to public and has all recent ones icluded

**Comment by Wolfie57** [comment]
Hey evilbudz, check out the following . . 
https://www.geonet.org.nz/data/types/eq_catalogue
http://api.geonet.org.nz/
Cheers
Power outage today so no chance for a fiddle, & still no eq data

**Comment by evilbudz** [comment]
am going to try and pass (curl_setopt($ch, CURLOPT_URL, 'http://wfs.geonet.org.nz/geonet/ows?service=WFS&version=1.0.0&request=GetFeature&typeName=geonet:quake_search_v1&maxFeatures=50&outputFormat=json) instead of  of the (curl_setopt($ch, CURLOPT_URL, 'https://www.geonet.org.nz/quakes/services/all.json)

**Comment by evilbudz** [comment]
seems to work except it display NaN for hours min and seconds lol

**Comment by evilbudz** [comment]
here is what i have, works if use other curl call to the actual all.json file and also notice when call the other way it does not have the inesity

<div class="row">
			<div class="span12 time_since invisible">
				<center><font color="blue"><span class="hours"></span></font> Hours <font color="blue"><span class="minutes"></span></font> Minutes <font color="blue"><span class="seconds"></span></font> Seconds since the last Earth Quake</center>
			</div>

**Comment by evilbudz** [comment]
have you tried to load what i gave you? that should work if not maybe your hosting provider is blocking something

**Comment by evilbudz** [comment]
i am guessing i have to alter the JS file in order for it to use the new id's damn it this is all new to me and very easy stuff seems so hard lol
